### PR TITLE
Wire serializer

### PR DESCRIPTION
**Pull request by rogeralsing**
This PR adds the ability to replace the protobuf serialization with Wire.
This allows .NET systems to pass normal POCO objects w/o defining a proto contract.

cc @raskolnikoov 

In order to use, place this code in the start up of your nodes:

```csharp
Serialization.RegisterSerializer(new WireSerializer(), true);
```

It should be noted, that in it's current form, this comes with quite a bit of performance penalty compared to the default serializer.
The reason for this is that Wire needs to roundtrip throough a Stream when serializing/deserializing, which adds extra cost compared to the raw ByteString as protobuf uses.

I've added a copy of the RemoteBenchmark specific to the Wire serializer.
It shows a throughput of ~ 1.3 mil msg/sec compared to the ~2.4 mil msg/sec using the default Protobuf serializer.

### CODE REVIEW

**Comment by rogeralsing** [comment]
Seems like there is some flakyness in one of the receive timeout tests:
`ReceiveTimeoutTests.receive_timeout_received_within_expected_time`

File: src/Proto.Remote/RemoteProcess.cs [hunk 1]
```diff
@@ -46,19 +46,12 @@ private void Send(PID _, object msg)
             }
         }
 
-        public static void SendRemoteMessage(PID pid, object msg,int serializerId)
+        public static void SendRemoteMessage(PID pid, object msg, int serializerId)
         {
             var (message, sender, _) = Proto.MessageEnvelope.Unwrap(msg);
 
-            if (message is IMessage protoMessage)
```
Comment (rogeralsing) [inline src/Proto.Remote/RemoteProcess.cs]: Removing leftover type assertion for protobuf

**Comment by raskolnikoov** [comment]
Sweet!

**Comment by tomliversidge** [comment]
ReceiveTimeout tests could be flaky as they use the world renowned harbinger of doom Thread.Sleep 

Can increase the sleep values to be more slow and stable rather than fast and flaky

**Comment by raskolnikoov** [comment]
Is this ready?

**Comment by rogeralsing** [comment]
@raskolnikoov yes

**Comment by Jesse1205** [comment]
Why delete the wire-serializer branch? it looks good

**Comment by cpx86** [comment]
@Jesse1205 it was already merged :)

**Pull request merged**